src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -3150,6 +3173,8 @@ void processContextElement
     return; // Error already in responseP
   } 
 
+  cacheSemTake(__FUNCTION__, "Processing Context Element");
```

Comment:
Yes, I think this should be improved. Note that csub cache was created to improve efficiency... but now we have the paradox that due to csub cache (more precicely, to the semaphores of csub cache) we are loosing any advantage that csub cache may bring!